### PR TITLE
Add a uuid filter to GETs on the /groups/ endpoint

### PR DESCRIPTION
**Pull request by wcmitchell**
Signed-off-by: Chris Mitchell <cmitchel@redhat.com>

### CODE REVIEW

File: docs/source/specs/openapi.json [hunk 1]
```diff
@@ -215,6 +215,21 @@
...
+            "name": "uuid",
+            "in": "query",
+            "description": "A list of UUIDs to filter listed groups.",
+            "required": false,
+            "schema": {
+              "type": "array",
+              "items": {
+                "type": "string"
+              },
+              "collectionFormat": "csv"
```
Comment (coderbydesign) [inline docs/source/specs/openapi.json]: It looks like this is a 2.0 thing, not compatible with 3.0? https://swagger.io/docs/specification/describing-parameters/

Comment (wcmitchell) [reply docs/source/specs/openapi.json]: Oh, yep. Dang. I had hopes it'd help clarify, but I'll pull it out

File: rbac/management/group/view.py [hunk 2]
```diff
@@ -59,12 +59,19 @@ def username_filter(queryset, field, value):
         filtered_set = queryset.filter(**filters)
         return filtered_set | Group.platform_default_set()
 
+    def uuid_filter(queryset, field, values):
+        """Filter for group uuid lookup."""
+        filters = {f'{field}__in': values.split(',')}
```
Comment (coderbydesign) [inline rbac/management/group/view.py]: We may want a UUID validation here? If I give an invalid UUID as one of the values in the query param, I get a 500.

Comment (wcmitchell) [reply rbac/management/group/view.py]: 👍

File: rbac/management/group/view.py [hunk 3]
```diff
@@ -59,12 +59,19 @@ def username_filter(queryset, field, value):
...
+        filtered_set = queryset.filter(**filters)
+        return filtered_set
+
     name = filters.CharFilter(field_name='name', lookup_expr='icontains')
     username = filters.CharFilter(field_name='principals', method=username_filter)
+    uuid = filters.CharFilter(field_name='uuid', method=uuid_filter)
```
[the remark this answers is not in the document]
Comment (wcmitchell) [reply rbac/management/group/view.py]: I went ahead and threw this in. I don't think it'll affect things much as they are now, but it seems to be what django-filter recommends, and doesn't seem a negative in any way I can see.

File: rbac/management/group/view.py (hunk 2 above)
Comment (wcmitchell) [reply rbac/management/group/view.py]: Added in a simple check that'll throw back a 400 and an 'invalid uuid' message.

File: rbac/management/group/view.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (astrozzc) [reply rbac/management/group/view.py]: Right, in my case I have to use the self. And it also remove the annoying lint warning 😁

**Pull request merged**